### DISCUSSION TITLE
calc 2.14.2.1 release

### DISCUSSION THREAD

**Discussion by lcn2** [lcn2/calc#82] [Announcements]
The following are the changes in this release:

    Added cal/fnv_tool.cal, a calc resource file defining:

	find_fnv_prime(bits)
	deprecated_fnv0(bits,fnv_prime,string)
	fnv_offset_basis(bits,fnv_prime)
	fnv1a_style_hash(bits,fnv_prime,prev_hash,string)

    Fixed sorted order of cal/README.

<hr /><em>This discussion was created from the release <a href='https://github.com/lcn2/calc/releases/tag/v2.14.2.1'>calc 2.14.2.1 release</a>.</em>

**Comment by BilakshanP** [comment]
When would these releases be updated on apt? In Ubuntu even after doing ```--upgrade``` it shows me that ```calc is already the newest version (2.12.7.2-4)```, as I was once discussing with you some time ago about ```d2r``` and ```r2d```missing. Well, I also don't know any up-to-date repositories/mirrors for ```calc```, and ```calc``` being such a generic term doesn't even give any useful info while searching online (Alas!).

**Reply by lcn2** [reply]
Hello @BilakshanP and thanks for raising this issue. 

We were aware that some people involved in Debian distributions were taking calc production updates and releasing them into the Debian distribution. 

We don't know if whoever was doing that was spotted, or is simply behind and plans to get around to updates for calc on Debian soon. 

We haven't (yet) developed the facility to create non-RPM based Linux distributions such as an apt based distribution. We are also unfamiliar with the requirements and practices that Debian might require of distributors. 

Any ideas on who the release author is for the Debian calc that you have?  Perhaps that person could be contacted and encouraged to pick up our latest production release: that might be the fastest was to get the calc on Debian updated. 

If for some reason such people are no around or not available or no longer interested in doing the Debian release of calc:

Any ideas on what us needed for a apt based release?

And ideas on what is required for such updates to make it into Debian?

**Reply by BilakshanP** [reply]
Thanks for the follow-up @lcn2 (nice username btw).

Well. I will try to contact or at-least figure out the Debian's release author.

In the mean time (or maybe as a complete substitute, I (we) may add a build script for various distros, but it could clutter the repo)

I am also not very much familiar with apt based releases, though I would like to see what could be done about this issue.

I'd like to hear your opinion.

**Reply by lcn2** [reply]
> Thanks for the follow-up @lcn2 (nice username btw).

Thanks @Bilakshanp: LCN2 is my APRANET NIC handle going back to the 1970's: LCN0 and LCN1 were already taken by hardware devices at the time, so LCN2 was the next available with my initials.

> Well. I will try to contact or at-least figure out the Debian's release author.
> 
> 
> 
> In the mean time (or maybe as a complete substitute, I (we) may add a build script for various distros, but it could clutter the repo)
> 
> 
> 
> I am also not very much familiar with apt based releases, though I would like to see what could be done about this issue.
> 
> 
> 
> I'd like to hear your opinion.

We are not familiar with building Debian apt, nor do we know the requirements for inclusion in Debian.  Others in the past were updating Debian for us.  We do have a Raspberry Pi running Debian, but that would be ARM release and may not be suitable for your needs.

Let us know what you find.

**Reply by lcn2** [reply]
> In the mean time (or maybe as a complete substitute, I (we) may add a build script for various distros, but it could clutter the repo)

Well having to apt file (or files?) in the release isn't much clutter, but it will introduce many extra steps (and extra time) where the newly released calc source has to be fetched to a Debian system, the apt file(s) build, set back and added to the release.

Although we have been asked to build RPMs for the IBM mainframe as well, so maybe such steps need to be considered anyway?  On the other hand those steps for an IBM mainframe release do not need to happen exactly at the same time as the Linux RPM release.  After all, the Debian release were someone's automated process to introduce the apt file(s) a bit later, after they detected we had issued a formal production stable release.

We do not possess a Debian system running on x86.  We only have Debian on ARM (Raspberry Pi) and so we we did it, we would only be able to generate Debian ARM files.  Would the Debian folks want both x86 and ARM apt files?  Probably.  So we would have to build some x86 hardware and maintain Debian on that hardware for the purposes of release Debian apt file(s)?

Now we have two different Debian systems, one of which we do not have, two different hosts to maintain, plus double to steps (and ways to make a mistake if not fully automated) and a lot more time to produce a calc release.

> I am also not very much familiar with apt based releases, though I would like to see what could be done about this issue.
> 
> I'd like to hear your opinion.

Someone else had a system whereby if they (or their system) detected a production release of calc, they had a script to generate stuff for Debian.

We were asked to introduce tags into our releases with a special form so that this other person's process could update Debian.  We got the impression that they had an automated system to do the calc apt file(s).

This is why contacting the person who released the other Debian apt file(s) might be the easier path: assuming they still want to do it.

## UPDATE 0

Is there some GitHub action that might allow those Debian apt file(s) to be released for x86 and perhaps arm?

Ideas and suggestions are welcome @BilakshanP.

**Comment by BilakshanP** [comment]
@lcn2 apologies for the late response.

I was messing with the ```dpkg``` utility today and was able to build ```.deb``` packages for my system ```(arm64)``` though it has some issues i.e ```postinst```, ```postrm``` and ```prerm``` files etc and if I am successful in streamlining the process I'll let you know and we probably can try integrating it with GitHub actions after that.

**Comment by lcn2** [comment]
> @lcn2 apologies for the late response.
> 
> I was messing with the `dpkg` utility today and was able to build `.deb` packages for my system `(arm64)` though it has some issues i.e `postinst`, `postrm` and `prerm` files etc and if I am successful in streamlining the process I'll let you know and we probably can try integrating it with GitHub actions after that.

Good to hear, @BilakshanP.  Looking forward to any updates you may have.
